src/snprintf.c

```diff
@@ -39,6 +39,8 @@ rcutils_snprintf(char * buffer, size_t buffer_size, const char * format, ...)
 int
 rcutils_vsnprintf(char * buffer, size_t buffer_size, const char * format, va_list args)
 {
+  RCUTILS_CAN_RETURN_WITH_ERROR_OF(-1);
+
   if (NULL == format) {
     errno = EINVAL;
```

Comment:
@brawner a no-op for now, but here's another side-effect that we're missing.

Reply:
I've added a new macro, `RCUTILS_CAN_FAIL_WITH` and `RCUTILS_FAULT_INJECTION_MAYBE_FAIL` for this condition.